pkg/elb/cleanup.go

```diff
@@ -199,6 +199,26 @@ func getServiceLoadBalancer(ctx context.Context, ec2API awsapi.EC2, elbAPI Descr
 	return &lb, nil
 }
 
+func getIngressELBName(ctx context.Context, metadataName string, hosts []string) string {
+	// Expected e.g. bf647c9e-default-appingres-350b-1622159649.eu-central-1.elb.amazonaws.com where AWS ALB name is
+	// bf647c9e-default-appingres-350b (cannot be longer than 32 characters).
+	hostNameParts := strings.Split(hosts[0], ".")
+	if len(hostNameParts[0]) == 0 {
+		logger.Debug("%s is ALB Ingress, but probably not provisioned or something other unexpected, skip", metadataName)
+		return ""
+	}
+	name := strings.TrimPrefix(hostNameParts[0], "internal-") // Trim 'internal-' prefix for ALB DNS name which is not a part of name.
+	idIdx := strings.LastIndex(name, "-")
+	if (idIdx) != -1 {
+		name = name[:idIdx] // Remove the ELB ID and last hyphen at the end of the hostname (ELB name cannot end with a hyphen)
+	}
+	if len(name) > 32 {
+		name = name[:32]
+	}
```

Comment:
Do we need this if we already remove the ELB ID?

Reply:
That's a good question. From my research ELB document [1] explicitly mentioned the rule when giving the name of load balancer, I think we still don't expect there's an ELB name exceeds 32 characters.

It might be better revise this part as exception case and throw the error directly, and let `getIngressLoadBalancer ` handle proper action (like skip ALB deletion right now, or interrupt the clean up process in the future).

Update: https://github.com/eksctl-io/eksctl/pull/7029/commits/4989bf86801145276785f5691c310e808f33cf12#diff-bf8761431813288f0320030b76870e9d4a40dd0e80e1ca725ef262184c498b90R214-R215

[1] https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_CreateLoadBalancer.html#API_CreateLoadBalancer_RequestParameters